### PR TITLE
SDIT-843 Allow update attendance in wrong prison or allocation ended

### PR DESCRIPTION
**Pull request by mikehalmamoj**
But prevent these if creating an attendance

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/AttendanceService.kt [hunk 1]
```diff
@@ -143,11 +150,8 @@ class AttendanceService(
     offenderBooking: OffenderBooking,
     bookingId: Long,
   ) =
-    offenderProgramProfileRepository.findByCourseActivityCourseActivityIdAndOffenderBookingBookingIdAndProgramStatusCode(
-      courseSchedule.courseActivity.courseActivityId,
-      offenderBooking.bookingId,
-      "ALLOC",
-    )
+    offenderProgramProfileRepository.findByCourseActivityAndOffenderBooking(courseSchedule.courseActivity, offenderBooking)
+      .maxByOrNull { it.startDate }
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/AttendanceService.kt:154]: As we can udpate if allocation ended don't restrict to status ALLOC - just get the latest

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/OffenderProgramProfile.kt [hunk 2]
```diff
@@ -39,7 +39,7 @@ data class OffenderProgramProfile(
   var program: ProgramService,
 
   @Column(name = "OFFENDER_START_DATE")
-  val startDate: LocalDate? = null,
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/OffenderProgramProfile.kt:42]: This column cannot be null even though table column is nullable. So point in wasting time catering for null values.

Comment (andymarke) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/OffenderProgramProfile.kt:42]: and I guess you never read this column and need to read old bad NOMIS data?

**Review by andymarke — APPROVED** [review]
lgtm

**Pull request merged**